### PR TITLE
Check if Event Has Conflicts

### LINKED ISSUE #84: Add buffers around event start and end time

### LINKED ISSUE #98: Incorporate code to check if event is conflicting w/popup
Integrate the code that asks if an event is conflicting with popup. This includes 
1. Calling the real function to check if an event has a conflict. 
2. Showing the conflicting event in the popup.

### LINKED ISSUE #108: When you modify an event, it shows up as conflicting with itself

### PR DESCRIPTION
**Pull request by allysmith12**
This PR implements the front end AND backend to check if a proposed event has conflicts and then the user can either continue editing the event or schedule it anyway.

FRONT END: 
1. Moves Event Conflict from Alert to Modal for more customization 
2. Calls backend to check if event has conflict
3. Shows list conflicting events in modal. If the event only conflicts due to to travel time, a blue (Travel Time) is next to the name. 

BACK END: 
1. Algorithm first checks if the proposed event directly overlaps with any of the other pod events. If so, it returns a list of all overlapping events. 
2. If the event has no direct conflict, the algorithm checks the travel time between the event directly before and after the proposed event. If the travel time > the time in between the events, the route marks the events as conflicting. 

Multiple Conflicting Events 
Screen Shot 2020-12-15 at 12 51 47 PM

Conflicting Event with Travel Time
Screen Shot 2020-12-15 at 12 37 11 PM

### CODE REVIEW

**Comment by allysmith12** [comment]
TO DO -- incorporate with Zulfis code once that is pushed, and then show the conflicting events in Alert

File: api/src/routes/events.ts [hunk 1]
```diff
@@ -70,8 +79,7 @@ eventRouter.put("/", [auth], async (req: Request, res: Response) => {
   }
 
   const eventId = req.body.id;
-  const id = (req as AuthRequest).user.id;
```
Comment (caromurp) [inline api/src/routes/events.ts:73]: is line 73 supposed to be removed?

Comment (allysmith12) [reply api/src/routes/events.ts:73]: Not sure, but I dont think so? @zusoomro

Comment (caromurp) [reply api/src/routes/events.ts:73]: other than this, I think it looks great! Also, it looks like the id variable from line 73 is never used so it's probably unnecessary

**Review by caromurp — APPROVED** [review]
LGTM!

File: api/src/routes/events.ts (hunk 1 above)
Comment (zusoomro) [reply api/src/routes/events.ts:73]: Yeah it was just something random that was throwing an error in my ide so i removed it whooops

**Review by zusoomro — APPROVED** [review]

**Pull request merged**